src/vector/layer.rs

```diff
@@ -467,6 +470,42 @@ pub trait LayerAccess: Sized {
             gdal_sys::OGR_L_SetAttributeFilter(self.c_layer(), null_mut());
         }
     }
+
+    /// Read batches of columnar [Arrow](https://arrow.apache.org/) data from OGR
+    ///
+    /// Refer to the example provided in `read_ogr_arrow.rs`.
+    ///
+    /// This API is new as of GDAL 3.6.
+    ///
+    /// # Safety
+    /// This uses the Arrow C Data Interface to operate on raw pointers provisioned from Rust.
+    /// These pointers must be valid and provisioned according to the ArrowArrayStream spec
+    #[cfg(all(major_is_3, minor_ge_6))]
+    unsafe fn read_arrow_stream(&mut self, out_stream: *mut gdal_sys::ArrowArrayStream) {
+        unsafe {
+            let version_check = gdal_sys::GDALCheckVersion(3, 6, std::ptr::null_mut());
+            if version_check == 0 {
+                panic!("Need GDAL>=3.6 for Arrow support");
```

Comment:
I don't think we need to panic here. The loader will prevent the program from running if the symbol is not available.